crates/next-api/src/middleware.rs

```diff
@@ -236,36 +253,97 @@ impl MiddlewareEndpoint {
             }]
         };
 
-        let edge_function_definition = EdgeFunctionDefinition {
-            files: file_paths_from_root,
-            wasm: wasm_paths_to_bindings(wasm_paths_from_root),
-            assets: paths_to_bindings(all_assets),
-            name: "middleware".into(),
-            page: "/".into(),
-            regions,
-            matchers,
-            env: this.project.edge_env().owned().await?,
-        };
-        let middleware_manifest_v2 = MiddlewaresManifestV2 {
-            middleware: [("/".into(), edge_function_definition)]
-                .into_iter()
-                .collect(),
-            ..Default::default()
-        };
-        let middleware_manifest_v2 = VirtualOutputAsset::new(
-            node_root.join("server/middleware/middleware-manifest.json".into()),
-            AssetContent::file(
-                FileContent::Content(File::from(serde_json::to_string_pretty(
-                    &middleware_manifest_v2,
-                )?))
-                .cell(),
-            ),
-        )
-        .to_resolved()
-        .await?;
-        output_assets.push(ResolvedVc::upcast(middleware_manifest_v2));
+        if matches!(config.runtime, NextRuntime::NodeJs) {
+            let chunk = self.node_chunk().to_resolved().await?;
+            let mut output_assets = vec![chunk];
+            if this.project.next_mode().await?.is_production() {
+                output_assets.push(ResolvedVc::upcast(
+                    NftJsonAsset::new(*this.project, *chunk, vec![])
+                        .to_resolved()
+                        .await?,
+                ));
+            }
+            let middleware_manifest_v2 = MiddlewaresManifestV2 {
+                middleware: [].into_iter().collect(),
+                ..Default::default()
+            };
```

Comment:
This manifest is completely empty. Is that intentional?

Reply:
Yes, node middleware uses different manifest which is generated separate

Reply:
But does the JS-side of Next.js really need this empty file then?

Reply:
Yea it's always expected to be present also makes it more clear we didn't just fail to generate a manifest.